File: Mushka.WebApi/ClientModels/Supplier/SupplierModel.cs
using System;
using System.Collections.Generic;

namespace Mushka.WebApi.ClientModels.Supplier
{
    public class SupplierModel
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public string Address { get; set; }

        public string Phone { get; set; }

        public string Email { get; set; }

        public string WebSite { get; set; }

        public string Notes { get; set; }

        public IEnumerable<SupplierContactPersonModel> ContactPersons { get; set; }
    }
}

File: frontend/src/app/shared/models/supplier.model.ts
import { ContactPerson } from './contact-person.model';
import { PaymentCard } from './payment-card.model';

export class Supplier {
  id: string;
  name: string;
  address: string;
  email: string;
  webSite: string;
  notes: string;
  contactPersons: ContactPerson[] = [];
  service: string;
  suppliesCount: number;
  paymentCards: PaymentCard[] = [];

  constructor(data: any) {
    Object.assign(this, data);
  }
}
